tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/PublicAssetMethodContract.scala

```diff
@@ -0,0 +1,628 @@
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.{BeforeEach, Test}
+import play.api.libs.json.Json
+
+import scala.jdk.CollectionConverters._
+
+object PublicAssetMethodContract {
+  case class UploadResponse(blobId: String, contentType: String, size: Long)
+}
+
+trait PublicAssetMethodContract {
+  @BeforeEach
+  def setUp(server: GuiceJamesServer): Unit = {
+    server.getProbe(classOf[DataProbeImpl])
+      .fluent()
+      .addDomain(DOMAIN.asString)
+      .addUser(BOB.asString(), BOB_PASSWORD)
+      .addUser(ANDRE.asString(), ANDRE_PASSWORD)
+
+    requestSpecification = baseRequestSpecBuilder(server)
+      .setAuth(authScheme(UserCredential(BOB, BOB_PASSWORD)))
+      .addHeader(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .build()
+      .log().ifValidationFails()
+  }
+
+  private def uploadAsset(content: Array[Byte] = "Your asset content here".getBytes,
+                          contentType: String = "image/png"): UploadResponse = {
+    val uploadResponse: String = `given`()
+      .basePath("")
+      .contentType(contentType)
+      .body(content)
+    .when
+      .post(s"/upload/$ACCOUNT_ID")
+    .`then`
+      .extract
+      .body
+      .asString
+
+    val blobId: String = (Json.parse(uploadResponse) \ "blobId").as[String]
+    val size: Long = (Json.parse(uploadResponse) \ "size").as[Long]
+    val assetType: String = org.apache.james.mailbox.model.ContentType.of((Json.parse(uploadResponse) \ "type").as[String])
+      .mimeType().asString()
+    UploadResponse(blobId, assetType, size)
+  }
+
+  @Test
+  def createValidRequestShouldReturnSuccessResponse(): Unit = {
+    val uploadResponse: UploadResponse = uploadAsset()
+
+    val request: String =
+      s"""{
+         |  "using": ["urn:ietf:params:jmap:core", "com:linagora:params:jmap:public:assets"],
+         |  "methodCalls": [
+         |    [
+         |      "PublicAsset/set", {
+         |        "accountId": "$ACCOUNT_ID",
+         |        "create": {
+         |          "4f29": {
+         |            "blobId": "${uploadResponse.blobId}"
+         |          }
+         |        }
+         |      }, "0"
+         |    ]
+         |  ]
+         |}""".stripMargin
+
+    val response: String = `given`()
+      .body(request)
+    .when()
+      .post()
+    .`then`
+      .statusCode(HttpStatus.SC_OK)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    assertThatJson(response)
+      .inPath("methodResponses[0]")
+      .isEqualTo(
+        s"""[
+           |    "PublicAsset/set",
+           |    {
+           |        "accountId": "$ACCOUNT_ID",
+           |        "oldState": "$${json-unit.ignore}",
+           |        "newState": "$${json-unit.ignore}",
+           |        "created": {
+           |            "4f29": {
+           |                "id": "$${json-unit.ignore}",
+           |                "blobId": "${uploadResponse.blobId}",
```

Comment:
As far as I can say there is supposed to be a blob copy, right? (otherwise upload cleanup would kill the PublicAsset). Correct?

We need to return here the blobId of the stored entity. 

Not the one used for the creation of the entity.

Reply:
If the blobId here is the stored entity, that means it is an internal value. I don't think the client can benefit from it =>  we should remove blobId from the response.

Reply:
> If the blobId here is the stored entity, that means it is an internal value. I don't think the client can benefit from it => we should remove blobId from the response.

I am unsure.

It could be nice as a user to be able to visualize my public asset and/or reuse its content. I think a JMAP blobId could still be meaningful.

But sure in a first time `blobId` is not fully compulsory in the response and might be removed from the model without (much) harm.

Reply:
Do you think the client will be confused with the property name "blobId" ?

- Upload asset X => return blobId="uploads-111"
- JMAP PublicAsset/set create: request blobId="uploads-111". And the response blobId ="abc-xyz"